Copy build/lib

ddn0 opened the pull request.

witchel commented on libtsuba/src/RDG.cpp:
```diff
@@ -871,3 +871,62 @@ galois::Result<void> tsuba::DropEdgeProperty(RDG* rdg, int i) {
+                        int64_t length) {
+
+  if (offset < 0 || length < 0) {
+    return tsuba::ErrorCode::InvalidArgument;
```
Why is offset a signed quantity?  With system APIs (like mmap), it usually is unsigned.
I mean, there isn't a good logical reason for length to be signed, but npos was -1, at least in one implementation I saw.  Maybe that is an arrow thing?

scober replied:
offset and length are signed to match the corresponding arguments in arrow::Table::Slice since this is supposed to replace that call in some sense. I don't know why they are signed in arrow, I think their typing choices are strange in a lot of places.

witchel commented on libtsuba/tests/fo_bench.cpp:
```diff
@@ -325,11 +451,43 @@ void ReadWriteFO(std::shared_ptr<arrow::Table> table, std::string bucket,
+    break;
+  case 1:
+    recovered = ReadPartial_v1(path, offset, length);
+    break;
```
It is nice to have the code to build the partial table in both ways.

scober replied:
My thought was this benchmark could hold historical versions of this logic so we have a baseline.

scober commented:
@witchel Do you have any thoughts on unit testing `LoadPartialTable` without exposing it? It would be a helper function internal to RDG.cpp if I had some other way of testing it.

witchel commented:
> @witchel Do you have any thoughts on unit testing `LoadPartialTable` without exposing it? It would be a helper function internal to RDG.cpp if I had some other way of testing it.

I put some S3 functions in the tsuba::internal namespace.  I'm not sure if that will work for you, but I think it sends a clear message.

scober commented:
I’ll check that out, that sounds like a good option.

ddn0 committed:
build: Copy machine-independent build artifacts too

ddn0 committed:
build: Do not build latex docs

Latex is not a common document format these days. Avoid requiring a
latex install to generate documentation.

ddn0 committed:
Fix missing include
